File: backend/calApp/calorie-service/src/main/java/com/caliq/calorie_service/models/response/ReportResponse.java
package com.caliq.calorie_service.models.response;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;
public record ReportResponse(List<DayAnalyse> anlyses){
    public record DayAnalyse(LocalDate date, BigDecimal weight,
                             BigDecimal sugarCons, BigDecimal sugarNorm,
                             BigDecimal fiberCons, BigDecimal fiberNorm,
                             BigDecimal kcalCons, BigDecimal kcalNorm,
                             BigDecimal fatCons, BigDecimal fatNorm,
                             BigDecimal proteinCons, BigDecimal proteinNorm,
                             BigDecimal carbsCons, BigDecimal carbsNorm,
                             BigDecimal waterCons, BigDecimal waterNorm){
    }
}

File: androidApp/NytriSync/app/src/main/java/com/ga3t/nytrisync/data/model/ReportResponse.kt
package com.ga3t.nytrisync.data.model
import java.math.BigDecimal
data class ReportResponse(
    val anlyses: List<DayAnalyse>
) {
    data class DayAnalyse(
        val date: String,
        val weight: BigDecimal?,
        val sugarCons: BigDecimal,
        val sugarNorm: BigDecimal,
        val fiberCons: BigDecimal,
        val fiberNorm: BigDecimal,
        val kcalCons: BigDecimal,
        val kcalNorm: BigDecimal,
        val fatCons: BigDecimal,
        val fatNorm: BigDecimal,
        val proteinCons: BigDecimal,
        val proteinNorm: BigDecimal,
        val carbsCons: BigDecimal,
        val carbsNorm: BigDecimal,
        val waterCons: BigDecimal,
        val waterNorm: BigDecimal
    )
}